packages/stylelint-config/index.js

```diff
@@ -98,7 +98,7 @@ module.exports = {
 		'selector-attribute-operator-space-before': 'never',
 		'selector-attribute-quotes': 'always',
 		'selector-class-pattern': [
-			'^([a-z][a-z0-9]*)(-[a-z0-9]+)*$',
+			'^([a-z][a-z0-9]*)(-[a-z0-9]+)*((__([a-z][a-z0-9]*)(-[a-z0-9]+)*)?(--([a-z][a-z0-9]*)(-[a-z0-9]+)*)?)$',
```

Comment:
I suggest to also set the [`resolveNestedSelectors ` option](https://stylelint.io/user-guide/rules/selector-class-pattern#resolvenestedselectors-true--false-default-false) and the message needs an update too. See [this example](https://github.com/wearerequired/coding-standards/blob/68ce2de9a3a723146afb3bb681d363ae869a7388/packages/stylelint-config/index.js#L31-L38) for inspiration.